### PR TITLE
Fix Event Emission Issue on Sync Step Failure

### PR DESCRIPTION
**Pull request by aminsato**
@omerfirmak 

have a look omer

#1370

### CODE REVIEW

**Review by joshklop — COMMENTED** [review]
I believe Omer's issue also refers to the call to `OnSyncStepDone(OpVerify...` about 10 lines above.

**Comment by aminsato** [comment]
> I believe Omer's issue also refers to the call to `OnSyncStepDone(OpVerify...` about 10 lines above.

I've made some changes.
Is that ok?

**Comment by omerfirmak** [comment]
Yeah, both `OnSyncStepDone(OpVerify` and `OnSyncStepDone(OpStore` should be put behind a `err == nil` check

**Comment by aminsato** [comment]
> Yeah, both `OnSyncStepDone(OpVerify` and `OnSyncStepDone(OpStore` should be put behind a `err == nil` check

Changed

File: db/event_listener.go [hunk 1]
```diff
@@ -1,17 +1,15 @@
 package db
 
-import "time"
-
 type EventListener interface {
-	OnIO(write bool, duration time.Duration)
+	OnIO(write bool)
```
Comment (omerfirmak) [inline db/event_listener.go:4]: why is this commit reverted?

File: sync/sync.go [hunk 2]
```diff
@@ -167,10 +166,11 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 				s.log.Warnw("Sanity checks failed", "number", block.Number, "hash", block.Hash.ShortString())
 				resetStreams()
 				return
+			} else {
+				s.listener.OnSyncStepDone(OpVerify, block.Number, time.Since(verifyTimer))
```
Comment (omerfirmak) [inline sync/sync.go:170]: this should be moved back to it's original position, before the `return func()` statement.

File: sync/sync.go [hunk 3]
```diff
@@ -184,7 +184,10 @@ func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stat
 				}
 				resetStreams()
 				return
+			} else {
```
Comment (omerfirmak) [inline sync/sync.go:187]: you dont need the else statement here. `if err != nil` branch returns early anyways.

File: sync/sync.go (hunk 2 above)
Comment (aminsato) [reply sync/sync.go:170]: but  @joshklop commented in [this](https://github.com/NethermindEth/juno/pull/1387#pullrequestreview-1702853868
) to change s.listener.OnSyncStepDone(OpVerify

Comment (omerfirmak) [reply sync/sync.go:179]: I think he meant something different. If you move this line to here, `time.Since(verifyTimer)` won't yield the correct value.

Comment (omerfirmak) [reply sync/sync.go]: We still need to check `if err == nil` before calling `OnSyncStepDone(OpVerify`

**Review by omerfirmak — APPROVED** [review]

**Pull request merged**